### DISCUSSION TITLE
Master branch vs open5GS_hss_cx branch

### DISCUSSION THREAD

**Discussion by abouillot-stl** [herlesupreeth/docker_open5gs#157] [Q&A] [answered]
Hello,

What is the difference between the `master` branch and the `open5gss_hss_cx` branch?

Looking at the timeline, the second evolved in parallel of the master since two years ago with no rebasing from master.

I guess the core components might be out of sync with the latest Open5GS and related components.

I see from the `docker-compse.yaml` file there is no FHoSS part, which might be good, given the ease to add subscribers to it. But there is the `hss`, which expose the `CX` port, sufficient to manage all IMS config items?

Is there somewhere some instructions on the specifics for this configuration?

Or maybe this branch, which seem still maintained, is more of an experiment than meant for global use.

Thanks

**Comment by s5uishida** [comment]
Hi @abouillot-stl 

You may try 4G VoLTE and SMS over IMS on the `open5gs_hss_cx` branch.

I don't know if Open5GS HSS(Cx) supports other IMS features.

I think `docker_open5gs` is a great tool to easily try 4G and 5G experimentally.

**Comment by herlesupreeth** [comment, marked as answer]
> What is the difference between the master branch and the open5gss_hss_cx branch?

master branch uses FHoSS for Cx interface towards IMS whereas open5gss_hss_cx uses open5gs HSS for Cx towards IMS. open5gs HSS have a bare minimum IMS configuration required to have a VoLTE call and SMS over IMS. Apart from that if you want to have custom IFCs etc. then you would have to dig into the code and change there (no GUI).

Due to minimal configuration option in open5gs HSS for Cx interface, FHoSS is used in master branch
